tests/loader_layer_tests.cpp

```diff
@@ -172,59 +177,344 @@ TEST_F(MetaLayers, InvalidComponentLayer) {
+                            "regular_test_layer.json");
+    env->add_explicit_layer(
+        ManifestLayer{}
+            .set_file_format_version(ManifestVersion(1, 1, 2))
+            .add_layer(ManifestLayer::LayerDescription{}.set_name(meta_layer_name).add_component_layers({regular_layer_name})),
+        "meta_test_layer.json");
+
+    uint32_t extension_count = 0;
+    EXPECT_EQ(VK_SUCCESS, env->vulkan_functions.vkEnumerateInstanceExtensionProperties(meta_layer_name, &extension_count, nullptr));
+    EXPECT_EQ(extension_count, 0);
+    {  // layer is not enabled
+        InstWrapper inst{env->vulkan_functions};
+        FillDebugUtilsCreateDetails(inst.create_info, env->debug_log);
+        inst.CheckCreate();
+        ASSERT_TRUE(env->debug_log.find(std::string("Meta-layer ") + meta_layer_name + " component layer " + regular_layer_name +
+                                        " adding device extension " + device_ext_name));
+
+        auto phys_dev = inst.GetPhysDev();
+        uint32_t extension_count = 0;
+        std::array<VkExtensionProperties, 1> extensions;
+        EXPECT_EQ(VK_SUCCESS,
+                  env->vulkan_functions.vkEnumerateDeviceExtensionProperties(phys_dev, meta_layer_name, &extension_count, nullptr));
+        EXPECT_EQ(extension_count, 1);  // return device_ext_name
+
+        EXPECT_EQ(VK_SUCCESS, env->vulkan_functions.vkEnumerateDeviceExtensionProperties(phys_dev, meta_layer_name,
+                                                                                         &extension_count, extensions.data()));
+        EXPECT_EQ(extension_count, 1);
+        EXPECT_TRUE(string_eq(extensions[0].extensionName, device_ext_name));
+    }
+    {  // layer is enabled
+        InstWrapper inst{env->vulkan_functions};
+        inst.create_info.add_layer(meta_layer_name);
+        FillDebugUtilsCreateDetails(inst.create_info, env->debug_log);
+        inst.CheckCreate();
+        ASSERT_TRUE(env->debug_log.find(std::string("Meta-layer ") + meta_layer_name + " component layer " + regular_layer_name +
+                                        " adding device extension " + device_ext_name));
+        auto phys_dev = inst.GetPhysDev();
+
+        uint32_t extension_count = 0;
+        std::array<VkExtensionProperties, 1> extensions;
+        EXPECT_EQ(VK_SUCCESS,
+                  env->vulkan_functions.vkEnumerateDeviceExtensionProperties(phys_dev, meta_layer_name, &extension_count, nullptr));
+        EXPECT_EQ(extension_count, 1);  // return device_ext_name
+
+        EXPECT_EQ(VK_SUCCESS, env->vulkan_functions.vkEnumerateDeviceExtensionProperties(phys_dev, meta_layer_name,
+                                                                                         &extension_count, extensions.data()));
+        EXPECT_EQ(extension_count, 1);
+        EXPECT_TRUE(string_eq(extensions[0].extensionName, device_ext_name));
+    }
+}
+// Override meta layer missing disable environment variable still enables the layer
```

Comment:
Sorry, this isn't clear to me.  Why do we want to enable a meta-layer missing the disable?  Or is this because you are forcing it as an explicit layer (which it seems like you are)?  If that's the case, I would call it more like ForceExplicit or something else.

Reply:
This test makes sure that the LunarG Override layer doesn't get enabled if it shouldn't, since the override layer is a bit special.